OpenRA.Mods.Common/Traits/BotModules/PowerDownBotMoudle.cs

```diff
@@ -0,0 +1,148 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Manages AI powerdown.")]
+	public class PowerDownBotMoudleInfo : ConditionalTraitInfo
+	{
+		[Desc("Delay (in ticks) between toggle")]
+		public readonly int Interval = 150;
+
+		[Desc("ConditionToken that powerdown")]
+		public readonly string PowerdownCondition = "powerdown";
+
+		public override object Create(ActorInitializer init) { return new PowerDownBotMoudle(init.Self, this); }
+	}
+
+	public class PowerDownBotMoudle : ConditionalTrait<PowerDownBotMoudleInfo>, IBotTick
```

Comment:
Module

Reply:
..........That is too embrassing

Reply:
The file is still called PowerDownBotMoudle.cs?

Reply:
Yeah but change in my version. still things to fix.

Reply:
fixed